src/components/leftSidebar/DynamicDropdown.tsx

```diff
@@ -0,0 +1,138 @@
+
+// Function to reset the dropdown options based on an input flag
+export function resetDropDownOptions(useFirstOptionOnly: boolean = false) {
+    // Clear the current array
+    dropDownOptions.length = 0;
+
+    if (useFirstOptionOnly || showDefaultAttributes == false) {
+        // Reset to only the first default option
+        if (DEFAULT_DROPDOWN_OPTIONS.length > 0) {
+            dropDownOptions.push(DEFAULT_DROPDOWN_OPTIONS[0]);
+        }
+        console.debug("DropDownOptions reset to only the first default option.");
+    } else {
+        // Reset to the full default options
+        dropDownOptions.push(...DEFAULT_DROPDOWN_OPTIONS);
+        console.debug("DropDownOptions reset to default.");
+    }
+}
+
+export function addDropDownOption(option: Option) {
+    // Check if an option with the same name or label already exists
+    const exists = dropDownOptions.some(
+        (existingOption) => existingOption.name === option.name || existingOption.label === option.label,
+    );
+
+    // Add the option only if it does not exist
+    if (!exists) {
+        dropDownOptions.push(option);
+        console.debug(`DropDownOption '${option.name}' added.`);
+    } else {
+        console.warn(`Option '${option.name}' already exists in dropDownOptions.`);
+    }
+}
+
+export default function DynamicDropdown({ options, onClick }: DropdownProps) {
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+    const [open, setOpen] = useState(false);
+    const [inputDropdownValue, setInputDropdownValue] = useState<string | undefined>(options[0]?.name);
+    const [value, setValue] = useState<AutocompleteValue<Option, false, false, false> | null>(options[0] || null);
+
+    // useEffect(() => {
+    //     // Call onClick with the default label if you'd like to trigger it on mount
+    //     onClick(options[0].label);
+    // }, [onClick, options]);
+
+    function handleClick(event: React.MouseEvent<HTMLElement>) {
+        if (open) {
+            setOpen(false);
+
+            if (anchorEl) {
+                anchorEl.focus();
+            }
+
+            setAnchorEl(null);
+        } else {
+            setAnchorEl(event.currentTarget);
+            setOpen(true);
+        }
+    }
+
+    const handleChange: SDSAutocompleteOnChange<Option, false, false, false> = (
+        _: React.SyntheticEvent<Element, Event>,
+        newValue: AutocompleteValue<Option, false, false, false>,
+    ) => {
+        setOpen(false);
+        setValue(newValue);
+
+        if (newValue) {
+            setInputDropdownValue(newValue.name.substring(0, 14)); // only print the first 14 characters to prevent overflow
+            onClick(newValue.name);
+        } else {
+            setInputDropdownValue(undefined);
+        }
+    };
+
+    function handleClickAway() {
+        if (open) {
+            setOpen(false);
+        }
+    }
+
+    return (
+        <div>
+            <InputDropdown
```

Comment:
From the SDS docs, I'd expect to see [`DropDown`](https://sds.czi.design/009eaf17b/p/42bdf2-dropdown-menu/b/177008) here instead of `InputDropdown` and `DropdownMenu`.

But I've never actually used any of those, so probably ignore?

Reply:
I've likewise never used these components, but my reading is the same. I think `DropDown` may accomplish much of what you're doing here already,